### PR TITLE
Enable client_repo in AK for provisioning

### PR DESCRIPTION
**Pull request by Gauravtalreja1**
Due to 6.14 [RFE](https://bugzilla.redhat.com/show_bug.cgi?id=1265120)  custom products are default disabled in AK/Host (being tested by @ColeHiggins2), so here for provisioning we need to override custom client repo to be enabled prior to provisioning, so host param like enable-puppet5 could install puppet-agent itself while provisioning.

### CODE REVIEW

**Comment by Gauravtalreja1** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_provisioning_puppet.py -k test_host_provisioning_with_external_puppetserver
provisioning: true

File: pytest_fixtures/component/provision_pxe.py [hunk 1]
```diff
@@ -124,6 +124,12 @@ def module_provisioning_rhel_content(
         environment=module_lce_library,
     ).create()
 
+    # Ensure client repo is enabled in the activation key
+    content = ak.product_content(data={'content_access_mode_all': '1'})['results']
+    client_repo_label = [i['label'] for i in content if i['name'] == client_repo.name][0]
```
Comment (adarshdubey-star) [inline pytest_fixtures/component/provision_pxe.py:129]: Non-blocking:
```suggestion
    client_repo_label = [repo['label'] for repo in content if repo['name'] == client_repo.name][0]
```

**Comment by adarshdubey-star** [comment]
Please add some description for the change.

File: pytest_fixtures/component/provision_pxe.py (hunk 1 above)
Comment (Gauravtalreja1) [reply pytest_fixtures/component/provision_pxe.py]: Updated, thanks!

**Comment by Gauravtalreja1** [comment]
PRT run 2821 passed for el7 and el8 provisioning, and el9 provisioning failed due to the common issue which @shubhamsg199 is investigating already, so this PR is RFR.

**Review by adarshdubey-star — APPROVED** [review]
Ack

**Review by vsedmik — APPROVED** [review]

**Review by jyejare — APPROVED** [review]

**Pull request merged**